internal/pkg/db/memory/addressables_test.go

```diff
@@ -0,0 +1,737 @@
+/********************************************************************************
+ *  Copyright 2020 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package memory
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+
+	"github.com/google/uuid"
+)
+
+// allTestAddressables a slice containing all the pre-constructed addressables for testing.
+var allTestAddressables = []contract.Addressable{addressable1, addressable2, addressable3}
+
+// addressable1 is a pre-constructed addressable with unique properties which does not match any other addressable when
+//  invoking functions which can return multiple addressables such as GetAddressablesByPort
+var addressable1 = contract.Addressable{
```

Comment:
Commented once; occurs elsewhere.  

Don't use global variables for your tests.  Think about creating a single factory function that returns a structure with random data to ensure variance across calls.